src/main.ts

```diff
@@ -62,13 +62,50 @@ export const loop = ErrorMapper.wrapLoop(() => {
 
         // Attempt to spawn a screep!
         let spawn = spawns[0];
-        spawn.spawnCreep([WORK, CARRY, MOVE], 'Worker1');
+        spawn.spawnCreep([WORK, MOVE, CARRY, MOVE], 'Worker1');
 
       } else {
         // It wasn't greater than 0... meaning we have no spawns.
         // This room can't create creeps.
       }
 
+      // get a reference to our creep
+      var mycreep = Game.creeps["Worker1"];
+
+      // What if my creep doesn't exist?
+      if (!mycreep) {
+        // LAME.
+        return;
+      }
+      mycreep.say("I live!");
+
+      // Find all the sources in the room.
+      // Typically two, sometimes 1, center rooms have 4, and non-player rooms have none.
+      let sources = room.find(FIND_SOURCES);
+
+      // Let's make sure there is atleast 1
+      if (sources.length > 0) {
+
+        // We can farm! We'll choose the first one.
```

Comment:
if there's more than one source in a room how do we tell it to choose the closest one?

Reply:
This is a great next step.  We could use a more advanced find function that finds the closest by range from the source to the controller using code like the following:

`room.controller.pos.findClosestByRange(FIND_SOURCES);`